include/xtensor/xstorage.hpp

```diff
@@ -644,7 +645,11 @@ namespace xt
         T* m_capacity = std::end(m_data);
 
         // stack allocated memory
-        alignas(XSHAPE_ALIGNMENT) T m_data[N > 0 ? N : 1];
+    #if XALIGNMENT != 0
+        alignas(XALIGNMENT) T m_data[N > 0 ? N : 1];
+    #else
+        T m_data[N > 0 ? N : 1];
+    #endif
```

Comment:
I think the alignment stuff could be entirely defined with a macro like this:

```cpp
#if XALIGNMENT != 0
    #define XTENSOR_ALIGNED alignas(XALIGNMENT)
#else
    #define XTENSOR_ALIGNED
#endif
```

Thus you can just declare `XTENSOR_ALIGNED T m_data[N > 0 ? N : 1];`. This will avoid to repeat the "if - else - endif" stuff in future aligned members.

Reply:
I actually changed it so that it takes the alignment from the allocator. it still requires some macro if else stuff because MSVC handles `alignas(0)` well, but doesn't handle `alignas(alignof(T))` well, and GCC don't seem to work with `alignas(0)` but `alignas(alignof(T))` is no problem ...